hepstats/hypotests/fitutils/sampling.py

```diff
@@ -0,0 +1,62 @@
+from .api_check import is_valid_pdf
+
+"""
+Module providing basic sampling methods.
+"""
+
+
+def base_sampler(models, nevents, floating_params=None):
+    """
+    Creates a samplers from models.
+
+    Args:
+        models (list): models to sample
+        nevents (list): number of in each sampler
+        floating_params (list): floating parameter in the samplers
+    """
+
+    assert all(is_valid_pdf(m) for m in models)
+    assert len(nevents) == len(models)
+
+    if floating_params:
+        floating_params = [f.name for f in floating_params]
+
+    samplers = []
+    fixed_params = []
+    for m in models:
+        def to_fix(p):
+            if floating_params:
+                return p.name in floating_params
+            else:
+                return False
+        fixed = [p for p in m.get_dependents() if not to_fix(p)]
+        fixed_params.append(fixed)
+
+    for i, (m, p) in enumerate(zip(models, fixed_params)):
+        sampler = m.create_sampler(n=nevents[i], fixed_params=p)
+        samplers.append(sampler)
+
+    return samplers
+
+
+def base_sample(samplers, ntoys, parameter=None, value=None):
+    """
+    Sample from samplers.
+
+    Args:
+        samplers (list): generators of samples
+        ntoys (int): number of samples to generate
+        parameter (optional): floating parameter in the sampler
```

Comment:
The usage of parameter and value really need to be further explained here ... unless you are an expert, you won't know what to use these for.

Reply:
Apologies if I seem picky.

Reply:
Ok that is linked to the other comments in `FrequentistCalculator.sample`. But here maybe I should let it as it is, in order to let the user design their own `base_sample` method if needed.

Reply:
That is fine, but it cannot harm to say these things in the doc.